documentation/Architecture/Overview.md

```diff
@@ -2,80 +2,105 @@
 
 This document outlines the architecture of the Microsoft Performance Toolkit SDK.
 
-For more detailed information on how to create your own project using the SDK, please view [Creating your own project](../Using-the-SDK/Creating-your-project.md). 
+For more detailed information on how to create your own project using the SDK, please view [Creating an SDK Plugin C# Project](../Using-the-SDK/Creating-your-project.md). 
 
-Examples:
-- [Custom Data Source Example](../../samples/SimpleDataSource/SimpleCustomDataSource.cs).
-- [Custom Data Processor Example](../../samples/SimpleDataSource/SimpleCustomDataProcessor.cs). 
 
+## High-Level Interfaces
 
-# Microsoft Performance Toolkit SDK
+The Microsoft Performance Toolkit SDK was built to empower users to analyze arbitrary data source (such as `.etl` files, `.ctf` files, SQL server logs, etc). 
+Users can develop *SDK plugins* that contain logic for processing data sources into tables, which are sent to the *SDK driver* (such as 
+the [Windows Performance Analyzer](https://docs.microsoft.com/en-us/windows-hardware/test/wpt/windows-performance-analyzer)) to render. At the highest 
+level, it exposes the following interfaces:
 
-The Microsoft Performance Toolkit SDK was built to empower users to analyze any arbitrary data source (e.g. .etl, .ctf, etc). 
-Using the SDK, any data source (including large files) can be quickly processed to generate custom graphs and tables.
-Below is a brief overview of the structure of the SDK.
+<img src=".attachments/overview.svg" width="800">
 
 
-![](.attachments/ArchitectureOverview.png)
+The interfaces depicted in this diagram are
 
+1) The SDK takes as input *data sources*, such as file paths, from the *SDK driver*. The SDK may receive this input when, for example, the user asks 
+the driver to open a file.
+2) The SDK passes the data sources to all *SDK plugins* which can support them. Plugins are 
+loaded dynamically at runtime by the SDK driver before the data sources are given to the SDK. In this example, 2 out of the 3 loaded plugins advertise that 
```

Comment:
I wonder if we need to reword this one?

plugins are loaded by the SDK's runtime which the driver orchestrates

Reply:
The driver is what calls `ProcessAssemblies` (or LoadPlugins with the newer plugins loader), so isn't saying "by the SDK driver" technically correct?